pkgs/development/python-modules/txtai/default.nix

```diff
@@ -0,0 +1,145 @@
+{ lib
+, buildPythonPackage
+, pythonOlder
+, fetchFromGitHub
+, pythonRelaxDepsHook
+# propagated build input
+, faiss
+, torch
+, transformers
+, huggingface-hub
+, numpy
+, pyyaml
+, regex
+# optional-dependencies
+, aiohttp
+, fastapi
+, uvicorn
+# , apache-libcloud
+, rich
+, duckdb
+, pillow
+, networkx
+, python-louvain
+, onnx
+, onnxruntime
+, soundfile
+, scipy
+, ttstokenizer
+, beautifulsoup4
+, nltk
+, pandas
+, tika
+, imagehash
+, timm
+, fasttext
+, sentencepiece
+, accelerate
+, onnxmltools
+, annoy
+, hnswlib
+, scikit-learn
+, sentence-transformers
+, croniter
+, openpyxl
+, requests
+, xmltodict
+# native check inputs
+, unittestCheckHook
+}:
+let
+  version = "5.5.1";
+  api = [ aiohttp fastapi uvicorn ];
+  # cloud = [ apache-libcloud ];
+  console = [ rich ];
+  database = [ duckdb pillow ];
+  graph = [ networkx python-louvain ];
+  model = [ onnx onnxruntime ];
+  pipeline-audio = [ onnx onnxruntime soundfile scipy ttstokenizer ];
+  pipeline-data = [ beautifulsoup4 nltk pandas tika ];
+  pipeline-image = [ imagehash pillow timm ];
+  pipeline-text = [ fasttext sentencepiece ];
+  pipeline-train = [ accelerate onnx onnxmltools onnxruntime ];
+  pipeline = pipeline-audio ++ pipeline-data ++ pipeline-image ++ pipeline-text ++ pipeline-train;
+  similarity = [
+    annoy
+    fasttext
+    hnswlib
+    # pymagnitude-lite
+    scikit-learn
+    sentence-transformers
+  ];
+  workflow = [
+    # apache-libcloud
+    croniter
+    openpyxl
+    pandas
+    pillow
+    requests
+    xmltodict
+  ];
+  all = api ++ console ++ database ++ graph ++ model ++ pipeline ++ similarity ++ workflow;
+  optional-dependencies = {
+    inherit api console database graph model pipeline-audio pipeline-image
+      pipeline-text pipeline-train pipeline similarity workflow all;
+  };
+in
+buildPythonPackage {
+  pname = "txtai";
+  inherit version;
+  format = "setuptools";
```

Comment:
```suggestion
  inherit version;
  
  format = "setuptools";
```

Reply:
I think she meant add a line to separate both.

Reply:
Using rec is fine, the overhead is minimal. We usually have pname, version and format in three adjacent lines.

Reply:
I personally dont like rec because its not immediately obvious which attributes are reused and which ones arent. If possible, i would like to leave it without rec.